### PR TITLE
Dockerfile and docker-compose.yml changes

### PR DESCRIPTION
**Pull request by ethomas2**
* Switch to multi-stage dockerfile
* Remove postgres dockerfile
* Remove Godeps (Godeps is deprecated)
* Add dependencies to Gopkg.toml
* Add vendor

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
-    volumes:
-      - ./docker_db/pgdata:/pgdata
-    ports:
-      - "5432"
-    environment:
-      - POSTGRES_USER=docker
-      - POSTGRES_PASSWORD=docker
-      - PGDATA=/pgdata
+    db:
+        image: postgres:9.6
```
Comment (zingales) [inline docker-compose.yml]: nice! i like not needed an extra dockerfile!

File: docker-compose.yml [hunk 2]
```diff
@@ -1,30 +1,24 @@
...
-      - TOKEN_SIGNING_KEY=AllYourBase
-    build:
-      context: ./
-      args:
-        app_env: ${APP_ENV} 
-    volumes:
-      - .:/go/src/github.com/atmiguel/cerealnotes/
-    ports:
-      - "8080:8080"
-    links:
```
Comment (zingales) [inline docker-compose.yml:30]: I think having the links/depends on is important. It makes sure docker starts the db image before the server image.

**Review by zingales — CHANGES_REQUESTED** [review]
As we talked about offline. I don't want vendor checked in.

File: docker-compose.yml (hunk 2 above)
Comment (ethomas2) [reply docker-compose.yml:30]: Done

**Review by zingales — APPROVED** [review]

**Pull request merged**